### PR TITLE
Deprecate Gerstner, move to FFT

### PR DESCRIPTION
**Pull request by huwb**
Gerstner still works but warning box displayed in Inspector, and documentation moved to FFT.

Perhaps a bold move but worth it in my opinion to narrow our focus, simplify Crest for users, and reduce code size later. LMK what you think.

### CODE REVIEW

**Comment by daleeidd** [comment]
I went and did a review of FFTs vs Gerstners and I missed something important in #903. It changes the spectrum quite a bit:

Screenshot (321)
WavesModerate before the PR

Screenshot (322)
WavesModerate after the PR

The spectrums might need adjusting too. It seems the higher the SAMPLES_PER_WAVE the more the spectrum needs boosting at the lower end to bring detail back.

**Comment by daleeidd** [comment]
Definitely bold but FFTs being used in examples is inevitable and not hard to remove deprecation if needed. I guess people will give feedback if anything is missing from FFTs.

**Comment by huwb** [comment]
Crap i missed that too. hm need to think about that..

File: docs/about/history.rst [hunk 1]
```diff
@@ -55,6 +55,12 @@ Performance
 
    -  Only calculate inverse view projection matrix when required.
 
+Deprecated
+^^^^^^^^^^
+.. bullet_list::
+
+   -  Made *ShapeGerstner* and *ShapeGerstnerBatched* components obsolete, *ShapeFFT* should is preferred. Example scenes moved over to *ShapeFFT*.
```
[the remark this answers is not in the document]
Comment (huwb) [reply docs/about/history.rst]: oops thanks

File: docs/user/wave-conditions.rst [hunk 2]
```diff
@@ -15,22 +15,14 @@ The following sections describe how to define the wave conditions.
...
-There are a few different systems that generate waves in Crest:
-
--  *ShapeGerstnerBatch* - the original system that manually adds together 'Gerstner' waves.
-   This component is inefficient, and does not support the 'wave spline' tech, and has lower quality waves.
-   It is about to be deprecated.
--  *ShapeGerstner* - high quality 'Gerstner'-based system.
-   Best choice for if running with *nographics* (no GPU).
--  *ShapeFFT* - high quality 'FFT'-based system. Best choice for maximum detail/realism.
-   Our innovative FFT tech supports waves at all scales and is also extremely efficient.
+The *ShapeFFT* component is uesd to generate waves in Crest.
```
[the remark this answers is not in the document]
Comment (huwb) [reply docs/user/wave-conditions.rst]: good catch, fixed

File: docs/user/collision-shape-for-physics.rst [hunk 3]
```diff
@@ -54,6 +54,10 @@ The helper classes always submit a fixed number of points this frame, so satisfy
 Gerstner Waves CPU
 ------------------
 
+.. important::
```
[the remark this answers is not in the document]
Comment (huwb) [reply docs/user/collision-shape-for-physics.rst]: oh cool thx

**Comment by huwb** [comment]
apologies for those doc fails! fixed now. think ill merge now, hope thats fine

**Pull request merged**